### PR TITLE
[CDS-668] chain build parrent link

### PR DESCRIPTION
**Pull request by Mowinski**
Add a link to parent build and top build in a chain.

cds-668

ToDo List:
- [x] adjust link position and appearance 
- [x] write tests

### CODE REVIEW

File: master/buildbot/db/buildrequests.py [hunk 1]
```diff
@@ -869,9 +869,11 @@ def thd(conn):
                     buildername=str(row.buildername),
                     number=row.number,
                     is_merged=bool(row.mergebrid),
+                    startbrid=row.startbrid,
                     full_name="{buildername} #{number} {merged}".format(merged=merged, **row),
                 )
                 buildrequests.append(buildrequest)
+            res.close()
```
Comment (campos-ddc) [inline master/buildbot/db/buildrequests.py]: This `close` should be in a `try: finally:` block (or using a context manager)

This reminds me, could you create a JIRA task to add these `close` calls everywhere? That would be part of fixing this problem:

https://unity.slack.com/archives/G2Y6YTCFL/p1518439105000191

File: www/templates/build.html [hunk 2]
```diff
@@ -21,6 +21,10 @@
 
             </div>
 
+            {% if parent_build_url %}
+                This build is child build of: <a href="{{ parent_build_url}}">this build</a>
```
Comment (campos-ddc) [inline www/templates/build.html]: Might be useful to also include a link to the top of the chain

**Review by campos-ddc — COMMENTED** [review]
Waiting on tests, let me know when the PR is actually finished (not sure how useful it is to have a PR with todos on it)

File: master/buildbot/status/web/build.py [hunk 3]
```diff
@@ -382,6 +384,14 @@ def content(self, req, cxt):
         cxt['chained_build'] = yield req.site.buildbot_service.master.db.buildrequests.getBuildChain(
             self.build_status.buildChainID,
         )
+        current_brids = self.build_status.brids[0]
```
Comment (warcholprzemo) [inline master/buildbot/status/web/build.py]: Is it ok that you use only first brid from brids?

**Review by warcholprzemo — APPROVED** [review]

File: master/buildbot/db/buildrequests.py (hunk 1 above)
Comment (warcholprzemo) [reply master/buildbot/db/buildrequests.py]: I created: https://jira.hq.unity3d.com/browse/CDS-1832

File: master/buildbot/status/web/build.py [hunk 4]
```diff
@@ -382,6 +384,19 @@ def content(self, req, cxt):
...
             self.build_status.buildChainID,
         )
+        current_brids = self.build_status.brids
+        current_build = next(ifilter(lambda x: x['id'] in current_brids, cxt['chained_build']), None)
+
+        cxt['parent_build_url'] = None
+        cxt['top_build_url'] = None
+
+        if current_build and current_build['startbrid']:
+            top_build = filter(lambda x: x['id'] == current_build['startbrid'], cxt['chained_build'])[0]
```
Comment (warcholprzemo) [inline master/buildbot/status/web/build.py]: Can we be sure that the filter returns a non-empty list?

**Review by warcholprzemo — APPROVED** [review]

**Review by campos-ddc — APPROVED** [review]

**Pull request merged**